packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,191 @@
+ * @example
+ * try {
+ *   await wallet.sendTransaction({
+ *     address: contractAddress,
+ *     abi,
+ *     functionName: "mintNFT",
+ *     args: [recipientAddress],
+ *   });
+ * } catch (e) {
+ *   if (e instanceof SendTransactionExecutionRevertedError) {
+ *     alert(`Transaction reverted: ${e.revertError.reason}`);
+ *   }
+ *   throw e;
+ * }
+ */
+export class EVMSendTransactionExecutionRevertedError extends EVMSendTransactionError {
+    constructor(
+        message: string,
+        public readonly viemError: BaseError,
+        revertError: ContractFunctionRevertedError,
+        public readonly txId: string | undefined,
+        public readonly stage: SendTransactionFailureStage,
+        public readonly data = revertError.data,
+        public readonly reason = revertError.reason,
+        public readonly signature = revertError.signature
+    ) {
+        super(message, viemError, stage, WalletErrorCode.SEND_TRANSACTION_EXECUTION_REVERTED);
+    }
+}
+
+export interface SendTransactionOptions {
+    /**
+     * The number of confirmations to wait for before yielding the transaction hash.
+     */
+    confirmations: number;
+    /**
+     * The timeout in milliseconds to wait for a transaction to confirm before throwing an error.
+     */
+    transactionConfirmationTimeout: number;
+}
+
+export class SendTransactionService {
+    constructor(
+        private publicClient: PublicClient,
+        private defaultSendTransactionOptions: SendTransactionOptions = {
+            confirmations: 2,
+            transactionConfirmationTimeout: 30_000,
+        }
+    ) {}
+
+    async sendTransaction(
+        request: TransactionServiceTransactionRequest,
+        client: SmartAccountClient<EntryPoint, Transport, Chain, SmartAccount<EntryPoint>>,
+        config: Partial<SendTransactionOptions> = {}
+    ): Promise<Hex> {
+        const { confirmations, transactionConfirmationTimeout } = this.getConfig(config);
+
+        // Simulate
+        await this.simulateCall(request, undefined, "simulation");
+
+        // Send
+        let hash;
+        try {
+            hash = await client.writeContract({
+                ...request,
+                account: client.account,
+                chain: client.chain,
+            });
+        } catch (e) {
+            if (e instanceof BaseError) {
+                throw new EVMSendTransactionError(e.message, e, "send");
+            }
+            throw e;
+        }
+
+        // Confirm
+        try {
+            const receipt = await this.publicClient.waitForTransactionReceipt({
+                hash,
+                confirmations,
+                timeout: transactionConfirmationTimeout,
+            });
+            return await this.handleReceipt(receipt, request);
+        } catch (e) {
+            if (e instanceof BaseError) {
+                throw new EVMSendTransactionError(e.message, e, "confirmation");
+            }
+            throw e;
+        }
+    }
```

Comment:
nit: are these comments needed?

`simulateCall`, `writeContract`, and `waitForTransactionReceipt` are probably descriptive enough